client.go

```diff
@@ -1044,6 +1044,41 @@ func (c *InternalHTTPClient) RowAttrDiff(ctx context.Context, index, frame strin
 	return rsp.Attrs, nil
 }
 
+// SendMessage posts a message synchronously.
+func (c *InternalHTTPClient) SendMessage(ctx context.Context, pb proto.Message) error {
+	msg, err := MarshalMessage(pb)
+	if err != nil {
+		return err
```

Comment:
in general, I'd like it if we were more consistent about adding context to errors as they're propagated up the call stack. This function returns errors in 4 different places, so doing something like `return fmt.Errorf("marshaling message: %v", err)` makes it much easier to understand where an error comes from when it pops out in a log message, or on the terminal. Otherwise you end up seeing "permission denied" and don't have a clue.